File: appinventor/components/src/com/google/appinventor/components/runtime/AccessibleComponent.java
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright 2020-2021 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

package com.google.appinventor.components.runtime;

public interface AccessibleComponent {

    //Sets the high contrast field of a component to be either True or False
    void setHighContrast(boolean isHighContrast);

    //Returns whether a component is in high contrast mode or not
    boolean getHighContrast();

    //Sets the large font field of a component to be either True or False
    void setLargeFont(boolean isLargeFont);

    //Returns whether a component is in large font mode or not
    boolean getLargeFont();
}

File: appinventor/components-ios/src/AccessibleComponent.swift
// -*- mode: swift; swift-mode:basic-offset: 2; -*-
// Copyright 2016-2023 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

import Foundation

@objc public protocol AccessibleComponent {
  /**
   * The component's contrast controller for better visibility
   */
  var HighContrast : Bool { get set }

  /**
   * The component's Font controller for better visibility
   */
  var LargeFont : Bool { get set }
}
